src/js/components/TextInput/StyledTextInput.js

```diff
@@ -49,6 +49,11 @@ Object.setPrototypeOf(StyledTextInput.defaultProps, defaultProps);
 const StyledTextInputContainer = styled.div`
   position: relative;
   width: 100%;
+
+  ${props =>
+    props.theme.textInput &&
+    props.theme.textInput.container &&
+    props.theme.textInput.container.extend};
```

Comment:
will you be willing to add a storybook example on `TextInput/stories/textinput.stories.js` that uses a custom theme that demonstrates the `extend` functionality? feel free to also add the `extend` examples for the StyledPlaceholder and the StyledSuggestions.
you can find a similar example on how it can be done here: https://github.com/grommet/grommet/blob/4d48520ceef31b779195761eea2226e4b33d2fbf/src/js/components/TextArea/textarea.stories.js#L25

Reply:
Hello, yup I can do so :)